src/charm.py

```diff
@@ -1,25 +1,110 @@
 #!/usr/bin/env python3
-# Copyright 2024 Ubuntu
+# Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
 
 """Charm the application."""
 
 import logging
 
 import ops
+from log import log_event_handler
+from state import State
 
 logger = logging.getLogger(__name__)
 
 
+CONTAINERS = [
```

Comment:
Could you do `list(COMMANDS.keys())` instead?

Reply:
Updated.